dissect/target/plugins/browsers/firefox.py

```diff
@@ -40,10 +48,27 @@ def __init__(self, target):
                 self.users_dirs.append((user_details.user, cur_dir))
 
     def check_compatible(self):
+        """Perform a compatibility check with the target.
+        This function checks if any of the supported browser directories
+        exists. Otherwise it should raise an ``UnsupportedPluginError``.
+
+        Raises:
+            UnsupportedPluginError: If the plugin could not be loaded.
+        """
         if not len(self.users_dirs):
             raise UnsupportedPluginError("No Firefox directories found")
 
-    def _iter_db(self, filename):
+    def _iter_db(self, filename: str) -> Iterator[SQLite3]:
+        """Generate a connection to a sqlite history database files.
+
+        Args:
+            filename: The filename as string of the database where the history is stored.
+        Yields:
+            connection to db_file (SQLite3)
+        Raises:
+            FileNotFoundError: If the history file could not be found.
+            SQLError: If the history file could not be opened.
```

Comment:
```suggestion
```
These aren't actually raised. This function catches those exceptions and only logs it.